smack-experimental/src/main/java/org/jivesoftware/smackx/mam/provider/MamFinProvider.java

```diff
@@ -0,0 +1,31 @@
+/**
+ *
+ * Copyright © 2015 Florian Schmaus
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mam.provider;
+
+/**
+ * MAM Fin Provider class.
+ * 
+ * @see <a href="http://xmpp.org/extensions/xep-0313.html">XEP-0313: Message
+ *      Archive Management</a>
+ *
+ */
+public class MamFinProvider {
```

Comment:
No problem, you are not in a race against the time. This will likely take some time to merge. In fact, I would welcome if could also write some integration tests for MAM.